src/middlewared/middlewared/common/smart/smartctl.py

```diff
@@ -27,6 +27,9 @@ async def get_smartctl_args(context, disk, smartoptions):
     if device is None:
         return
 
+    if device["vendor"].lower().strip() == "nvme":
```

Comment:
Why not combine this with the nvme trap on line 23?

Reply:
Because the `device` object is the dictionary with the `disk` details. The `disk` is just the name of it (`nvme`, `sd`, etc).